### PR TITLE
Make it explicit to install distro to get auto-instrumentation working

### PR DESCRIPTION
**Pull request by lonewolf3739**
# Description

Fixes https://github.com/open-telemetry/opentelemetry-python/issues/1931

### CODE REVIEW

File: opentelemetry-instrumentation/README.rst [hunk 1]
```diff
@@ -11,11 +11,14 @@ Installation
 
 ::
 
-    pip install opentelemetry-instrumentation
+    pip install opentelemetry-distro[otlp] opentelemetry-instrumentation
```
Comment (lzchen) [inline opentelemetry-instrumentation/README.rst:14]: I think we should separate out these sections. Right underneath this, an explanation like what you have in the note would be good (users need to install "A" distro), and then have the `pip install` follow that excerpt. An explanation that `opentelemetry-distro` contains the default distro would be good too. This way users won't be confused as to why or what the `opentelemetry-distro` packge is.

**Review by owais — APPROVED** [review]
This looks fine but I still think our default recommendation should be to just install the distro and run `opentelemetry-instrument` as part of getting started docs. It should be followed up with an advanced section that documents other packages and how one may use them without the distro. IMO `opentelemetry-distro` should be the only or the first package users install when getting started.

**Comment by lonewolf3739** [comment]
Completely agree. That will make a huge difference in terms of user experience for the first the time users. Meanwhile, I wanted to add this note since it is not obvious to someone that without a distro auto instrumentation doesn't produce any telemetry.

**Review by lzchen — APPROVED** [review]

**Comment by lzchen** [comment]
@ocelotl 
Quick ping to make sure this pr doesn't get overlooked during the migration of `opentelemetry-instrumentation` to core.

https://github.com/open-telemetry/opentelemetry-python/pull/1959

**Comment by ocelotl** [comment]
> @ocelotl
> Quick ping to make sure this pr doesn't get overlooked during the migration of `opentelemetry-instrumentation` to core if merged.
> 
> @lonewolf3739
> Can you fix the builds so we can merge this asap?
> 
> [open-telemetry/opentelemetry-python#1959](https://github.com/open-telemetry/opentelemetry-python/pull/1959)

Roger that, @lzchen.

**Review by ocelotl — APPROVED** [review]

**Comment by lonewolf3739** [comment]
^ @lzchen it's green now.

**Pull request merged**